api/nodes/views.py

```diff
@@ -1149,7 +1159,17 @@ def get_queryset(self):
         if not hasattr(node_addon, 'get_folders'):
             raise EndpointNotImplementedError('Endpoint not yet implemented for this addon')
 
-        return node_addon.get_folders(path=path, folder_id=folder_id)
+        #  Convert v1 errors to v2 as much as possible.
+        try:
+            return node_addon.get_folders(path=path, folder_id=folder_id)
+        except InvalidAuthError:
+            raise NotAuthenticated('This add-on could not be authenticated.')
+        except HTTPError as exc:
+            v2_exception = HTTP_CODE_MAP.get(exc.code)
+            if v2_exception:
+                raise v2_exception
+            else:
+                raise ServiceUnavailableError(detail='This add-on is unavailable.')
```

Comment:
How about "An unknown error occurred"? If we end up here, we can't say for certain that the provider is down.

Reply:
I think raising a `500` here (instead of `503`) is most correct.